### DISCUSSION TITLE
Storage - ability to specify different storage buckets

### DISCUSSION THREAD

**Discussion by nthonymiller** [firebase/firebase-js-sdk#4402] [Q&A] [answered]
Is it possible with this new API, to be able to specify a different storage bucket. 

I would like to be able to switch between various storage buckets, not just use the one specified in the firebase config initialization.

**Comment by samtstern** [comment, marked as answer]
@nthonymiller thanks!  This is definitely missing at the moment so I think we should turn this into an Issue rather than a Discussion because it's a bug.

Let's track this here:
https://github.com/firebase/firebase-js-sdk/issues/4407

**Reply by hsubox76** [reply]
I think we need to change the type definitions but you should be able to do:
```
const storage = getStorage(app, "gs://my-custom-bucket")
```
Feel free to comment further in the linked issue (#4407) if that doesn't work or anything else needs to be changed other than the type definition and jsdoc.

**Reply by samtstern** [reply]
I think the fact that they're not in the type definitions makes it seem like they don't exist (to me anyway).  I rely on autocomplete a lot.

**Reply by samtstern** [reply]
@nthonymiller see my comment here:
https://github.com/firebase/firebase-js-sdk/pull/4410#issuecomment-772817556

This changed in a recent `exp` release, although @hsubox76 is making it even clearer with a new docs PR.
